src/ILCompiler/src/Program.cs

```diff
@@ -170,6 +188,51 @@ private int Run(string[] args)
             return 0;
         }
 
+        private TypeDesc FindType(CompilerTypeSystemContext context, string typeName)
+        {
+            TypeDesc foundType = context.SystemModule.GetTypeByCustomAttributeTypeName(typeName);
+            if (foundType == null)
+            {
+                throw new CommandLineException($"Type '{typeName}' not found");
+            }
+            return foundType;
+        }
+
+        private MethodDesc CheckAndParseSingleMethodModeArguments(CompilerTypeSystemContext context)
+        {
+            if (_singleMethodName == null && _singleMethodTypeName == null && _singleMethodGenericArgs == null)
+                return null;
+
+            if (_singleMethodName == null || _singleMethodTypeName == null)
+                throw new CommandLineException("Both method name and type name are required parameters for single method mode");
```

Comment:
NUTC has `-singlemethodbyname TypeName MethodName` where TypeName is in the ECMA-335 SerString format and method name is just the method name, followed by a bunch of `-singleMethodGenericParam Param`.

This is the exact same thing, except we need to split `-singlemethodbyname TypeName MethodName` because System.CommandLine can't do a switch with two arguments.